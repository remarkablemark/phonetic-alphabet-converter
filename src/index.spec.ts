import converter, { NATO_PHONETIC_ALPHABET } from '.';

describe('error', () => {
  it.each([
    undefined,
    null,
    0,
    1,
    {},
    [],
    () => undefined,
    new Date(),
    Symbol()
  ])('throws when first argument is %s', value => {
    expect(() => {
      converter(value as string);
    }).toThrowError('First argument must be a string');
  });
});

describe('converter', () => {
  it.each([
    ['mark', ['mike', 'alpha', 'romeo', 'kilo']],
    ['abc', ['alpha', 'bravo', 'charlie']]
  ])('converts "%s" correctly', (text, expected) => {
    expect(converter(text)).toEqual(expected);
  });
});

describe('NATO_PHONETIC_ALPHABET', () => {
  it('matches snapshot', () => {
    expect(NATO_PHONETIC_ALPHABET).toMatchSnapshot();
  });
});
