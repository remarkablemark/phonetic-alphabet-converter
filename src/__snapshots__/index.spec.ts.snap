// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NATO_PHONETIC_ALPHABET matches snapshot 1`] = `
Object {
  "a": "alpha",
  "b": "bravo",
  "c": "charlie",
  "d": "delta",
  "e": "echo",
  "f": "foxtrot",
  "g": "golf",
  "h": "hotel",
  "i": "india",
  "j": "juliet",
  "k": "kilo",
  "l": "lima",
  "m": "mike",
  "n": "november",
  "o": "oscar",
  "p": "papa",
  "q": "quebec",
  "r": "romeo",
  "s": "sierra",
  "t": "tango",
  "u": "uniform",
  "v": "victor",
  "w": "whiskey",
  "x": "x-ray",
  "y": "yankee",
  "z": "zulu",
}
`;
